src/perform_step/linear_perform_step.jl

```diff
@@ -28,6 +28,42 @@ function perform_step!(integrator, cache::MagnusMidpointCache, repeat_step=false
   integrator.destats.nf += 1
 end
 
+function initialize!(integrator, cache::MagnusGauss4Cache)
+  integrator.kshortsize = 2
+  integrator.fsalfirst = cache.fsalfirst
+  integrator.fsallast = cache.k
+  resize!(integrator.k, integrator.kshortsize)
+  integrator.k[1] = integrator.fsalfirst
+  integrator.k[2] = integrator.fsallast
+  integrator.f(integrator.fsalfirst, integrator.uprev, integrator.p, integrator.t) # For the interpolation, needs k at the updated point
+  integrator.destats.nf += 1
+end
+
+function perform_step!(integrator, cache::MagnusGauss4Cache, repeat_step=false)
+  @unpack t,dt,uprev,u,p,alg = integrator
+  @unpack W,k,tmp = cache
+  mass_matrix = integrator.f.mass_matrix
+
+  L1 = integrator.f.f
+  update_coefficients!(L1,u,p,t+dt*(1/2+sqrt(3)/6))
+  L2 = integrator.f.f
```

Comment:
This isn't going to work: it's going to alias the two operators.

Reply:
@ChrisRackauckas So, if i allocate A = Matrix(L), then update L again and then assign B=Matrix(L), should it work then ?

Reply:
yes it should. We won't want to do this for performance reasons, but do it for now so we can start with a working version to then optimize.

Reply:
Doing it increased the order to 2.